### PR TITLE
module/atu: add driver module for address translation unit

### PR DESCRIPTION
**Pull request by shriram-k-arm**
The Address Translation Unit(ATU) provides means to convert the logical address (32 bit LA) from the MSCP memory space to the physical address residing in the system memory.

Add a driver module for the ATU that provides an interface for modules to request a translation region to be mapped, unmapped or output bus attributes to be configured during the runtime phase.

Change-Id: Ic9bf58cf75c74ecc2d760e041369f3df824e1efa

### CODE REVIEW

**Comment by tarek-arm** [comment]
Jenkins: OK to test

File: unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -106,7 +106,7 @@ list(APPEND FWK_SRC ${UNITY_SRC_ROOT}/unity.c)
 list(APPEND SCP_UNITY_SRC ${TEST_ROOT}/unity_mocks/scp_unity.c)
 
 #Append common unit tests below here
-list(APPEND UNIT_MODULE dvfs)
```
Comment (mohamedasaker-arm) [inline unit_test/CMakeLists.txt:109]: Please don't remove dvfs unit tests
list(APPEND UNIT_MODULE atu)
list(APPEND UNIT_MODULE dvfs)

**Review by mohamedasaker-arm — CHANGES_REQUESTED** [review]
Thanks for the change 👍🏼 
Just a minor thing fix!

File: unit_test/CMakeLists.txt (hunk 1 above)
Comment (shriram-k-arm) [reply unit_test/CMakeLists.txt:109]: Sorry, I missed this. Thanks for pointing out.

**Review by mohamedasaker-arm — APPROVED** [review]
Looks Great!

**Pull request merged**